### PR TITLE
 Add cluster autoscaler w/NAP test involving GPUs

### PR DESCRIPTION
**Pull request by losipiuk**
This is very trivial e2e tests verifying interoperability of Cluster Autoscaler with Node Auto Provisioning and GPUs.
More specific tests to come.

```release-note
NONE
```

### CODE REVIEW

File: test/e2e/autoscaling/cluster_size_autoscaling.go [hunk 1]
```diff
@@ -309,6 +312,26 @@ var _ = SIGDescribe("Cluster size autoscaling [Slow]", func() {
 		})
 	}
 
+	// TODO consider moving to [Feature:ClusterSizeAutoscalingGpu] as soon as NAP goes out of beta. Currently
+	// project needed to run the NAP tests require whitelisting for NAP alpha
+	It("NAP should add a pool with GPUs if unschedulable POD which require GPU exists [Feature:ClusterSizeAutoscalingScaleWithNAP]", func() {
+		framework.SkipUnlessProviderIs("gke")
+
+		installNvidiaDriversDaemonSet()
+		framework.ExpectNoError(enableAutoprovisioning(""))
```
[the remark this answers is not in the document]
Comment (MaciekPytel) [reply test/e2e/autoscaling/cluster_size_autoscaling.go]: 1. Agreed
2. Let's do that in separate PR

**Comment by MaciekPytel** [comment]
/lgtm cancel 
After comments by @aleksandra-malinowska

**Comment by losipiuk** [comment]
@aleksandra-malinowska updated

File: test/e2e/autoscaling/cluster_size_autoscaling.go [hunk 2]
```diff
@@ -309,6 +312,26 @@ var _ = SIGDescribe("Cluster size autoscaling [Slow]", func() {
 		})
 	}
 
+	// TODO consider moving to [Feature:ClusterSizeAutoscalingGpu] as soon as NAP goes out of beta. Currently
+	// project needed to run the NAP tests require whitelisting for NAP alpha
+	It("NAP should add a pool with GPUs if unschedulable POD which require GPU exists [Feature:ClusterSizeAutoscalingScaleWithNAP]", func() {
+		framework.SkipUnlessProviderIs("gke")
+
+		installNvidiaDriversDaemonSet()
+		framework.ExpectNoError(enableAutoprovisioning(`"resource_limits":{"name":"nvidia-tesla-k80", "minimum":0, "maximum": 3}`))
```
Comment (aleksandra-malinowska) [inline test/e2e/autoscaling/cluster_size_autoscaling.go]: This won't work without limits for CPU and memory, unfortunately:( (see enableAutoprovisioning implementation...)

**Comment by DirectXMan12** [comment]
Shouldn't these tests be running out of the cluster autoscaler repo?

**Comment by losipiuk** [comment]
> Shouldn't these tests be running out of the cluster autoscaler repo?

That would be definitely preferred. We should move tests to CA repo and setup automation so CA master is tested (currently we are testing latest CA release against k8s master).
That change is on our roadmap. Hopefully we will get to it eventually...

**Comment by DirectXMan12** [comment]
> Hopefully we will get to it eventually...

That should be made a priority for the next release.

**Pull request merged**